stl/inc/format

```diff
@@ -2557,8 +2557,215 @@ template <_Format_supported_charT _CharT, class _Traits>
+        _Ptr      = _Buf_data;
+        _Capacity = _Buf_capacity;
+    }
+
+    virtual void grow(size_t _Capacity) = 0;
+
+public:
+    using value_type      = _Ty;
+    using const_reference = const _Ty&;
+
+    _Fmt_buffer(const _Fmt_buffer&) = delete;
+    void operator=(const _Fmt_buffer&) = delete;
+
+    _Ty* begin() noexcept {
+        return _Ptr;
+    }
+    _Ty* end() noexcept {
+        return _Ptr + _Size;
+    }
+
+    const _Ty* begin() const noexcept {
+        return _Ptr;
+    }
+    const _Ty* end() const noexcept {
+        return _Ptr + _Size;
+    }
+
+    size_t size() const noexcept {
+        return _Size;
+    }
+
+    size_t capacity() const noexcept {
+        return _Capacity;
+    }
+
+    _Ty* data() noexcept {
+        return _Ptr;
+    }
+
+    const _Ty* data() const noexcept {
+        return _Ptr;
+    }
+
+    void clear() noexcept {
+        _Size = 0;
+    }
+
+    void try_resize(const size_t _Count) {
+        try_reserve(_Count);
+        _Size = _Count <= _Capacity ? _Count : _Capacity;
+    }
+
+    void try_reserve(const size_t _New_capacity) {
+        if (_New_capacity > _Capacity) {
+            grow(_New_capacity);
+        }
+    }
+
+    void push_back(const _Ty& _Value) {
+        try_reserve(_Size + 1);
+        _Ptr[_Size++] = _Value;
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(size_t) {}
+    size_t count() const {
+        return 0;
+    }
+    size_t limit(const size_t _Size) {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    size_t _Count = 0;
+    size_t _Limit;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(const size_t _Lim) : _Limit(_Lim) {}
+    size_t count() const {
+        return _Count;
+    }
+    size_t limit(const size_t _Size) {
+        size_t _Avail = _Limit > _Count ? _Limit - _Count : 0;
+        _Count += _Size;
+        return _Size < _Avail ? _Size : _Avail;
+    }
+};
+
+inline constexpr size_t _Fmt_buffer_size = 256;
+
+template <class _OutputIt, class _Ty, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_Ty> {
+private:
+    _OutputIt _Out;
+    _Ty _Data[_Fmt_buffer_size];
+
+protected:
+    void grow(size_t) final override {
+        if (this->size() == _Fmt_buffer_size) {
+            flush();
+        }
+    }
+    void flush() {
+        auto _Size = this->size();
+        this->clear();
+        _Out = _STD _Copy_unchecked(_Data, _Data + this->limit(_Size), _STD move(_Out));
+    }
+
+public:
+    explicit _Fmt_iterator_buffer(_OutputIt _Output, size_t _Size = _Fmt_buffer_size)
+        : _Traits(_Size), _Fmt_buffer<_Ty>(_Data, 0, _Fmt_buffer_size), _Out(_STD move(_Output)) {}
+    ~_Fmt_iterator_buffer() {
+        flush();
+    }
+
+    _OutputIt out() {
+        flush();
+        return _STD move(_Out);
+    }
+    ptrdiff_t count() const {
+        return static_cast<ptrdiff_t>(_Traits::count() + this->size());
+    }

```

Comment:
Should this function also return `size_t` as other `count()`s do?

Reply:
I think this is causing the test failure on signed/unsigned mismatch.

Reply:
It returns `ptrdiff_t` to avoid casts in multiple call sites. Internally everything operates on `size_t`, not sure if it's worth changing.